data/src/main/java/com/jordifierro/androidbase/data/repository/NoteDataRepository.java

```diff
@@ -27,57 +27,45 @@ public NoteDataRepository(RestApi restApi) {
     @Override
     public Observable<NoteEntity> createNote(UserEntity user, final NoteEntity note) {
         return this.restApi.createNote(user.getAuthToken(), note)
-                .map(new Func1<Response<NoteEntity>, NoteEntity>() {
-                    @Override
-                    public NoteEntity call(Response<NoteEntity> noteEntityResponse) {
-                        handleResponseError(noteEntityResponse);
-                        return noteEntityResponse.body();
-                    }
+                .map(noteEntityResponse -> {
+                    handleResponseError(noteEntityResponse);
+                    return noteEntityResponse.body();
                 });
     }
 
     @Override
     public Observable<NoteEntity> getNote(UserEntity user, int noteId) {
         return this.restApi.getNote(user.getAuthToken(), noteId)
-                .map(new Func1<Response<NoteEntity>, NoteEntity>() {
-                    @Override
-                    public NoteEntity call(Response<NoteEntity> noteEntityResponse) {
-                        handleResponseError(noteEntityResponse);
-                        return noteEntityResponse.body();
-                    }
+                .map(noteEntityResponse -> {
+                    handleResponseError(noteEntityResponse);
+                    return noteEntityResponse.body();
                 });
     }
 
     @Override
     public Observable<List<NoteEntity>> getNotes(UserEntity user) {
         return this.restApi.getNotes(user.getAuthToken())
-                .map(new Func1<Response<List<NoteEntity>>, List<NoteEntity>>() {
-                    @Override
-                    public List<NoteEntity> call(Response<List<NoteEntity>> listResponse) {
-                        handleResponseError(listResponse);
-                        return listResponse.body();
-                    }
+                .map(listResponse -> {
+                    handleResponseError(listResponse);
+                    return listResponse.body();
                 });
     }
 
     @Override
     public Observable<NoteEntity> updateNote(UserEntity user, NoteEntity note) {
         return this.restApi.updateNote(user.getAuthToken(), note.getId(), note)
-                .map(new Func1<Response<NoteEntity>, NoteEntity>() {
-                    @Override
-                    public NoteEntity call(Response<NoteEntity> noteEntityResponse) {
-                        handleResponseError(noteEntityResponse);
-                        return noteEntityResponse.body();
-                    }
+                .map(noteEntityResponse -> {
+                    handleResponseError(noteEntityResponse);
+                    return noteEntityResponse.body();
                 });
     }
 
     @Override
     public Observable<Void> deleteNote(UserEntity user, int noteId) {
```

Comment:
Another potential issue with the use case of deleting note is that after the note is deleted, `NoteDetailPresenter` will be resumed, which will fetch the updated note in `onResume()` method and get 404 error as the note has been deleted in `NoteEditPresenter`, even though the `NoteDetailPresenter` is finished immediately. We probably need to add a flag to indicate when a note is deleted, there is no need to fetch its latest content `onResume()`. Any thoughts on how to achieve this use case and still keep the architecture clean?